crates/pixi_build_frontend/src/tool/spec.rs

```diff
@@ -48,6 +60,82 @@ impl IsolatedToolSpec {
             ..self
         }
     }
+
+    /// Installed the tool in the isolated environment.
+    pub async fn install(
+        &self,
+        cache_dir: &Path,
+        context: ToolContext,
+    ) -> miette::Result<IsolatedTool> {
+        let gateway = Gateway::builder()
```

Comment:
This needs to because the in-memory cache is not shared otherwise.

Reply:
refactored into `ToolContextBuilder`